content/docs/rfcs/51/README.md

```diff
@@ -173,45 +169,103 @@ This example node is part of shards `13`, `14`, and `45` in the Status main-net
-Like with DHTs, the hash space is split into parts,
-each covered by a Pubsub topic (mesh network) that carries content topics which are mapped into the respective part of the hash space.
+Autosharding selects shards automatically and is the default behavior for shard choice.
+The other choices being static and named sharding as seen in previous sections.
+Shards (pubsub topics) SHOULD be computed from content topics with the procedure below.
+
+## Rendezvous Hashing
+
+Also known as the Highest Random Weight (H.R.W.) method, has many properties useful for shard selection.
+- Distributed agreement. Local only computation without communication.
+- Load balancing. Content topics are spread as equally as possible to shards.
+- Minimal disruption. Content topics switch shards infrequently.
+- Low overhead. Easy to compute.
+
+### Algorithm
+
+For each shard,
+hash using Sha2-256 the concatenation of
+the content topic `application` field (N U.T.F.-8 bytes),
+`version` (N U.T.F.-8 bytes),
+the `cluster` index (2 bytes) and
+the `shard` index (2 bytes)
+take the first 64 bits of the hash,
+divided by 2^64,
+compute the natural logarithm of this number then
+take the negative of the weight (default 1.0) divided by it.
+Finally, sort the shard value pairs.
+
+The shard with the highest value SHOULD be used.
+
+### Example
+| Field | Value | Hex |
+|---    |---    |---  |
+| `application`| "myapp"| 0x6d79617070|
+| `version`    | "1"    | 0x31
+| `cluster`    | 1      | 0x0001
+| `shard`      | 6      | 0x0006
+
+- SHA2-256 of `0x6d796170703100010006` is `0xfdac5fb315b791cca3ccde738ebb0b8d2742519fce1086f2a3d2409cb22a7dd2`
+- The first 64 bits `0xfdac5fb315b791cc` divided by `0xffffffffffffffff` is ~`0.99`
+- The natural logarithm of `0.99` is ~`-0.01`
+- `-1` divided by `-0.01` equals ~`99.5`
+- Shard 6 has the priority value 99.5
+
+## Content Topics Format for Autosharding
+Content topics MUST follow the format in [23/WAKU2-TOPICS](https://rfc.vac.dev/spec/23/#content-topic-format).
+In addition, 2 prefixes MAY be added to content topics.
+Generation & bias.
+When omitted default values are used.
+Generation default value is `0`.
+Bias default value is `unbiased`.
+
+- The full length format is `{generation}/{bias}/{application-name}/{version-of-the-application}/{content-topic-name}/{encoding}`
+- The short length format is `/{application-name}/{version-of-the-application}/{content-topic-name}/{encoding}`
+
+### Example
+
+- Full length `/0/unbiased/myapp/1/mytopic/cbor`
+- Short length `/myapp/1/mytopic/cbor`
+
+### Generation
+The generation number monotonously increases and indirectly refers to the total number of shards of the Waku Network.
+
+The first generation (zero) MUST use 8 shards in total.
+The cluster index is 1 and the indices of each shards are numbered 0 to 7.
+
+This document will be updated for future generations.
+
+### Bias
+Bias is used to skew the priority of shards via weights.
```

Comment:
I agree with @fryorcraken comment but I'd also add more detail re the expected values range for this field and what is the implication of having a lower or higher `bias` value.

Reply:
this bias thingy seems a bit weird to me, mind ellaborating more on why you want to bias this?

reading this

> bias is used to skew the priority values via the weighting function in a specific way.

> One of the requirement was to allow app the affect the sharding, to this end a simple 20/80 split between higher shards and lower shards has been added.The lower 20% of shards will be picked more often when adding the "lower20" bias in the topic and "higher80" would favor the other shards.

I get its meant to "bias" the shard selection, but imho we shouldnt have this feature. ofc anyone can "golf" the shard by appending stuff to the content topic (similar to vanity-eth with addresses) but imho this should be as obscure as possible, so that people just use the shard they are assigned to.

thin also from a app/user perspective. I set my app name, some topic nameand, "bias"? what is this? which number do i chose 1, 5, 100? content topic should abstract sharding away, and this bias introduces some coupling. more cognitive load to users for a feature that imho we dont need.

Reply:
> I'm all for removing any bias.

I think we can remove it (in fact, even with it removed, nothing would prevent us from experimenting with different ways of hashing ourselves, but I agree it's better to keep the specs clear of this).

> For what I understood of the requirements we did NOT want an algorithm with only one input 

From my side at least, the question was whether we should have a way to still optionally hardcode shard selection (not simply bias it) into the content topic, but we agreed elsewhere that this won't be needed for now. Other than that, I think our best approach is a simple hash of the app/version parts of the (single input) content topic.

Reply:
noice, lets remove the concept of bias then.

Reply:
deal!

Reply:
Is that the discussion that led to the introduction of _bias_ ? https://github.com/waku-org/nwaku/pull/1854#discussion_r1276975061

My statement still hold that while we can keep things obscure and just instruct dev to choose 1 _application_ value and whatever _name_ value for their content topic, I would also keep in our back pocket the ability to instruct an application to use different _application_ values if they need to scale beyond a shard capacity.